apps/fxc-front/src/app/logic/score/league/leagues.ts

```diff
@@ -1,22 +1,50 @@
-import { League } from '../league';
-import { CzechEurope, CzechLocal, CzechOutEurope } from './czech';
-import { FrCfd } from './frcfd';
-import { Leonardo } from './leonardo';
-import { UKXCLClub, UKXCLInternational, UKXCLNational } from './ukxcl';
-import { WXC } from './wxc';
-import { NorwayLeague, XContest, XContestPPG } from './xcontest';
+import { ScoringRules } from 'optimizer';
 
-export const LEAGUES: { [name: string]: League } = {
-  czl: new CzechLocal(),
-  cze: new CzechEurope(),
-  czo: new CzechOutEurope(),
-  fr: new FrCfd(),
-  leo: new Leonardo(),
-  nor: new NorwayLeague(),
-  ukc: new UKXCLClub(),
-  uki: new UKXCLInternational(),
-  ukn: new UKXCLNational(),
-  xc: new XContest(),
-  xcppg: new XContestPPG(),
-  wxc: new WXC(),
+// allowed league codes
+export const leagueCodes = ['czl', 'cze', 'czo', 'fr', 'leo', 'nor', 'ukc', 'uki', 'ukn', 'xc', 'xcppg', 'wxc'];
+export type LeagueCode = (typeof leagueCodes)[number];
+
+export const LEAGUES: Readonly<Record<LeagueCode, string>> = {
+  czl: 'Czech (ČPP local)',
+  cze: 'Czech (ČPP Europe)',
+  czo: 'Czech (ČPP outside Europe)',
+  fr: 'France (CFD)',
+  leo: 'Leonardo',
+  nor: 'Norway (Distanseligaen)',
+  ukc: 'UK (XC League, Club)',
+  uki: 'UK (XC League, International)',
+  ukn: 'UK (XC League, National)',
+  xc: 'XContest',
+  xcppg: 'XContest PPG',
+  wxc: 'World XC Online Contest',
 };
+
+export function getScoringRules(league: string): ScoringRules {
```

Comment:
`string` -> `LeagueCode`

you can also:

```ts

const LeagueRules = {
  czl: ScoringRules.CzechLocal,
  ...
};

export function getScoringRules(league: LeagueCode): ScoringRules {
  const rules = LeaguesRules[league);
  if (rules == null) {
    throw new Error(...);
  }
  return rules;
}
```

Reply:
Ok. Done